docs/source/guide/refactor_code/index.rst

```diff
@@ -1,7 +1,156 @@
+
+   # Create dummy variables encoding species
+   df["d_versicolor"] = df["variety"].apply(lambda x: 1 if x == "Versicolor" else 0)
+   df["d_virginica"] = df["variety"].apply(lambda x: 1 if x == "Virginica" else 0)
+
+   # View data
+   df
+
+   # Initiate the model
+   lm_2 = LinearRegression()
+
+   # Fit the model
+   lm_2.fit(
+      X=df[["petal.width", "d_versicolor", "d_virginica"]],
+      y=df["sepal.width"],
+   )
+
+   # Check result
+   print(f"Coefficient for petal.width: {lm_2.coef_[0]}")
+   print(f"Coefficient for d_versicolor: {lm_2.coef_[1]}")
+   print(f"Coefficient for d_virginica: {lm_2.coef_[2]}")
+
+   # Map each species to a color
+   color_map = {"Setosa": "green", "Versicolor": "blue", "Virginica": "red"}
+   df["variety_color"] = df["variety"].map(color_map)
+
+   # Plot petal vs. sepal width by species
+   df.plot.scatter("petal.width", "sepal.width", c="variety_color")
+   plt.show()
+
+   # Create interaction terms for species
+   df["i_versicolor"] = df["petal.width"] * df["d_versicolor"]
+   df["i_virginica"] = df["petal.width"] * df["d_virginica"]
+
+   # Initiate the model
+   lm_3 = LinearRegression()
+
+   # Fit the model
+   lm_3.fit(
+      X=df[["petal.width", "d_versicolor", "d_virginica", "i_versicolor", "i_virginica"]],
+      y=df["sepal.width"],
+   )
+
+   # Check result
+   print(f"Coefficient for petal.width: {lm_3.coef_[0]}")
+   print(f"Coefficient for d_versicolor: {lm_3.coef_[1]}")
+   print(f"Coefficient for d_virginica: {lm_3.coef_[2]}")
+   print(f"Coefficient for i_versicolor: {lm_3.coef_[3]}")
+   print(f"Coefficient for i_virginica: {lm_3.coef_[4]}")
+
+As shown, the code interweaves various plots, models, and print statements, reflecting dynamic nature
+of the development stage of data science work.
+
+Say we are interested in productionizing the second model (``lm_2``). Normally, this would involve manually
+sifting through the entire code to identify relevant parts only. We can cut through such manual labor with
+the help of LineaPy.
+
+First, we store the model as a LineaPy artifact:
+
+.. code:: python
+
+   # Save desired model as an artifact
+   artifact = lineapy.save(lm_2, "linear_model_v2")
+
+Then, we simply ask for its "sliced" code, like so:
+
+.. code:: python
+
+   # Get "sliced" code
+   print(artifact.get_code())
+
+And we get:
+
+.. code:: none
+
+   import pandas as pd
+   from sklearn.linear_model import LinearRegression
+
+   df = pd.read_csv(
+      "https://raw.githubusercontent.com/LineaLabs/lineapy/main/examples/tutorials/data/iris.csv"
+   )
+   df["d_versicolor"] = df["variety"].apply(lambda x: 1 if x == "Versicolor" else 0)
+   df["d_virginica"] = df["variety"].apply(lambda x: 1 if x == "Virginica" else 0)
+   lm_2 = LinearRegression()
+   lm_2.fit(
+      X=df[["petal.width", "d_versicolor", "d_virginica"]],
+      y=df["sepal.width"],
+   )
+
+which is more concise and manageable than what we initially had --- a long, messy collection of various operations.
```

Comment:
might also want to highlight that some of the stuff in the original development code didn't have any impact on the artifact we care about. just to make sure users understand that we're not rewriting code to condense it. we're just removing the extra stuff.